src/TrackGenerator.cpp

```diff
@@ -2036,3 +2049,38 @@ double TrackGenerator::getPhi(int azim) {
   else
     return 2 * M_PI - _phi[azim - _num_azim];
 }
+
+
+/**
+ * @brief Deletes the Timer's timing entries for each timed code section
+ *        code in the loop over track for ray tracing.
+ */
+void TrackGenerator::clearTimerSplits() {
+  _timer->clearSplits();
+}
+
+
+/**
+ * @brief Prints a report of the timing statistics to the console.
+ */
+void TrackGenerator::printTimerReport() {
+
+  std::string msg_string;
+
+  log_printf(TITLE, "TRACKGENERATOR TIMING REPORT");
+
+  /* Get the total runtime */
+  double tot_time = _timer->getSplit("Total time");
+  msg_string = "Total time for ray tracing";
+  msg_string.resize(53, '.');
```

Comment:
Sure, I just added a new `REPORT_WIDTH` macro to "constants.h" and dropped it into the timer reports in `Solver` and `TrackGenerator'. Perhaps a formalized "report" interface in the `log` module would be useful (and minimize code reuse) for these new timer / memory footprint reports you speak of @samuelshaner.